src/bentoml/_internal/frameworks/transformers.py

```diff
@@ -3,29 +3,74 @@
 import os
 import typing as t
 import logging
+import platform
+import warnings
 from types import ModuleType
-from typing import TYPE_CHECKING
 
 import attr
 
 import bentoml
-from bentoml import Tag
-from bentoml.models import Model
-from bentoml.models import ModelContext
-from bentoml.models import ModelOptions
-from bentoml.exceptions import NotFound
-from bentoml.exceptions import BentoMLException
-from bentoml.exceptions import MissingDependencyException
 
+from ..tag import Tag
 from ..types import LazyType
-from ..utils.pkg import find_spec
+from ..utils import LazyLoader
 from ..utils.pkg import get_pkg_version
-
-if TYPE_CHECKING:
-    from bentoml.types import ModelSignature
+from ..utils.pkg import pkg_version_info
+from ...exceptions import NotFound
+from ...exceptions import BentoMLException
+from ...exceptions import MissingDependencyException
+from ..models.model import Model
+from ..models.model import ModelContext
+from ..models.model import ModelOptions
+from ..models.model import ModelSignature
+
+if t.TYPE_CHECKING:
+    import cloudpickle
+    from transformers.models.auto.auto_factory import (
+        _BaseAutoModelClass as BaseAutoModelClass,
+    )
 
     from ..models.model import ModelSignaturesType
-    from ..external_typing import transformers as ext
+
+    class SimpleDefaultMapping(t.TypedDict, total=False):
+        pt: tuple[str, ...]
+        tf: tuple[str, ...]
+
+    class ModelDefaultMapping(t.TypedDict):
+        model: SimpleDefaultMapping
+
+    DefaultMapping = (
+        ModelDefaultMapping
+        | SimpleDefaultMapping
+        | dict[tuple[str, ...], ModelDefaultMapping]
+    )
+
+    TupleStr = tuple[str, ...]
+    TupleAutoModel = tuple[BaseAutoModelClass, ...]
+
+    class PreTrainedProtocol(t.Protocol):
+        def save_pretrained(self, save_directory: str, **kwargs: t.Any) -> None:
+            ...
+
+        @classmethod
+        def from_pretrained(
+            cls, pretrained_model_name_or_path: str, **kwargs: t.Any
+        ) -> PreTrainedProtocol:
+            ...
```

Comment:
Is t.Protocol supported in 3.7?

Reply:
Nope. t.Protocol is 3.8+ feature.

Reply:
Should we still use it then?

Reply:
When we drop 3.7 then I think we can move these out.